controllers/actions.summerwind.net/horizontal_runner_autoscaler_batch_scale.go

```diff
@@ -143,8 +143,51 @@ func (s *batchScaler) batchScale(ctx context.Context, batch batchScaleOperation)
 	}
 
 	copy := hra.DeepCopy()
+	now := time.Now()
+
+	if hra.Spec.MaxReplicas != nil && len(copy.Spec.CapacityReservations) > *hra.Spec.MaxReplicas {
```

Comment:
> The code in this PR has the added advantage of working properly when amount is a number other than +1 or -1.

@Nuru Hey! I was reviewing this once again and I think your statement and this line doesn't align with each other- shouldn't maxReplicas be compared against the sum of all the amount if you're going to support other amounts?

Reply:
NVM. Could you check https://github.com/actions/actions-runner-controller/pull/2502#discussion_r1169376047?